### DISCUSSION TITLE
Cargo build before running each test

### DISCUSSION THREAD

**Discussion by wcampbell0x2a** [sourcefrog/cargo-mutants#456] [Q&A]
My testing suite needs to run `cargo build` before every invocation of `cargo test`. Is there a provided way of doing this with `cargo-mutants`? I didn't see any. Thanks!

**Comment by sourcefrog** [comment]
It already does a `cargo build` before running tests, or more precisely `cargo test --no-run` which builds the tests (and other targets) without running them.

https://mutants.rs/how-it-works.html

Is that not working for you, or do you need to build some other non-default target?

There is also this open feature request to run a custom command before tests: https://github.com/sourcefrog/cargo-mutants/issues/430. It would be helpful to get more information or examples of how people would want to use it.

**Reply by wcampbell0x2a** [reply]
I specifically need to build binaries that are built in `package-cli`, but `package-test` uses. (There isn't a way in cargo to describe this relationship with bins). So I need something like `cargo build --bins`.

**Reply by sourcefrog** [reply]
Is it in the same workspace? We have `--test-workspace` but maybe we should have a separate `--build-workspace`. And/or just a generic additional build command.
